activerecord/lib/active_record/associations/association.rb

```diff
@@ -189,7 +189,10 @@ def extensions
       def load_target
         @target = find_target(async: false) if (@stale_state && stale_target?) || find_target?
         if !@target && set_through_target_for_new_record?
-          @target = through_association.target.association(reflection.source_reflection_name).target
+          @target = reflection.chain.reverse.drop(1).reduce(through_association.target) do |middle_target, reflection|
```

Comment:
I don't get what that `chain` is for. Isn't `chain` without argument essentially a noop?

Also not a big fan for the `.reverse.drop.reduce`, I think a more straightforward  `reverse_each` could be easier to understand.

Reply:
> I don't get what that chain is for. Isn't chain without argument essentially a noop?

In the through reflection case it returns the chain of nested through reflections, which we iterate over to keep setting the target until we get to the desired(last) one.
- https://github.com/rails/rails/blob/410ad1c8a18ebf7f544d77d2cb82de2a0e87d23a/activerecord/lib/active_record/reflection.rb#L332-L334
- https://github.com/rails/rails/blob/410ad1c8a18ebf7f544d77d2cb82de2a0e87d23a/activerecord/lib/active_record/reflection.rb#L1032-L1051

> Also not a big fan for the .reverse.drop.reduce, I think a more straightforward reverse_each could be easier to understand.

I opted for the `reduce` because the initial middle target is the current association target (`through_association.target`), which makes the first entry in the reversed chain redundant (hence the `drop(1)`). I agree that it's not easy to grok, here's an attempt at a refactor:

```ruby
          middle_target = through_association.target
          reflection.chain.reverse_each.with_index do |reflection, index|
            next if index == 0 # skip reflection for initial middle target
            break unless middle_target
            @target = middle_target = middle_target.association(reflection.source_reflection_name).target
          end
```

We only want to set `@target` to a truthy value if we get to the end of the chain. Let me know what you think and I can push this up if it reads easier.

Reply:
Ok, I just did some minor changes which I think improve readability.

Reply:
LGTM, thank you!